evennia/web/website/views/help.py

```diff
@@ -1,17 +1,149 @@
+    """
+    return getattr(help_entry, 'key', 'unknown_topic')
+
+
+def can_read_topic(cmd_or_topic, account):
+    """Check if an account or puppet has read access to a help entry.
+
+    Args:
+        cmd_or_topic (Command, HelpEntry or FileHelpEntry): The topic/command to test.
+        account: the account or puppet checking for access.
+
+    Returns:
+        bool: If command can be viewed or not.
+
+    Notes:
+        This uses the 'read' lock. If no 'read' lock is defined, the topic is assumed readable
+        by all.
+        Even if this returns False, the entry will still be visible in the help index unless
+        `can_list_topic` is also returning False.
+    """
+    if inherits_from(cmd_or_topic, "evennia.commands.command.Command"):
+        return cmd_or_topic.auto_help and cmd_or_topic.access(account, 'read', default=True)
+    else:
+        return cmd_or_topic.access(account, 'read', default=True)
 
 
-class HelpMixin(TypeclassMixin):
+def collect_topics(account):
+    """Collect help topics from all sources (cmd/db/file).
+
+    Args:
+        account (Character or Account): Account or Character to collect help topics from.
+
+    Returns:
+        cmd_help_topics (dict): contains Command instances.
+        db_help_topics (dict): contains HelpEntry instances.
+        file_help_topics (dict): contains FileHelpEntry instances
+    """
+
+    # collect commands of account and all puppets
+    # skip a command if an entry is recorded with the same topics, category and help entry
+    cmd_help_topics = []
+    if not str(account) == 'AnonymousUser':
+        # create list of account and account's puppets
+        puppets = account.db._playable_characters + [account]
+        # add the account's and puppets' commands to cmd_help_topics list
+        for puppet in puppets:
+            for cmdset in puppet.cmdset.get():
+                # removing doublets in cmdset, caused by cmdhandler
+                # having to allow doublet commands to manage exits etc.
+                cmdset.make_unique(puppet)
+                # retrieve all available commands and database / file-help topics.
+                # also check the 'cmd:' lock here
+                for cmd in cmdset:
+                    # skip the command if the puppet does not have access
+                    if not cmd.access(puppet, 'cmd'):
+                        continue
+                    # skip the command if the puppet does not have read access
+                    if not can_read_topic(cmd, puppet):
+                        continue
+                    # skip the command if it's help entry already exists in the topic list
+                    entry_exists = False
+                    for verify_cmd in cmd_help_topics:
+                        if verify_cmd.key and cmd.key and \
+                           verify_cmd.help_category == cmd.help_category and \
+                           verify_cmd.__doc__ == cmd.__doc__:
```

Comment:
Minor style comment - you can just put the whole thing in parentheses to break lines. The `\` to break lines is very rarely needed in Python. 

```python
if (verify_cmd.key and cmd.key
        and verify_cmd.help_category == cmd.help_category
        and verify_cmd.__doc__ == cmd.__doc__):
    entry_exists = True 
    break

Reply:
done